weave/trace/box.py

```diff
@@ -19,16 +19,31 @@ class BoxedInt(int):
     _id: int | None = None
     ref: Ref | None = None
 
+    def __deepcopy__(self, memo: dict) -> "BoxedInt":
+        res = BoxedInt(self)
+        memo[id(self)] = res
```

Comment:
where is `memo` defined? what is that?

Reply:
It's part of the python magic method definition:
https://docs.python.org/3/library/copy.html#object.__deepcopy__